### DISCUSSION TITLE
How Run  Python as sidecar without stopping it  with a tauri app

### DISCUSSION THREAD

**Discussion by smith0022** [tauri-apps/tauri#8135] [Q&A]
so basically i have a python script that should run in parallel with the tauri app ,the react side communicates with python script using websocket and this communication has to be continuous.so i tried to bundle it but then it executes the whole file and i want to make run in parallel
 the line of code where i call the sidecar :
const command = Command.sidecar("bin/python/test-x86_64-unknown-linux-gnu");
  const output = command.execute();

I'm using this in react js and added this line in the app.js file

**Comment by FabianLars** [comment]
Can you share a bit of your code (mainly interested in the part where you spawned the sidecar)?

**Reply by smith0022** [reply]
const command = Command.sidecar("bin/python/test-x86_64-unknown-linux-gnu");
const output = command.execute();
outside this function in app.js i imported the command first:
import { Command } from "@tauri-apps/api/shell";

**Reply by FabianLars** [reply]
if you call command.spawn() it will run in parallel. execute() is only meant to be used if you want to wait for it to finish.

**Comment by Tek-Ce** [comment]
On 30 Oct 2023 8:44 am, smith0022 ***@***.***> wrote:
so basically i have a python script that should run in parallel with the tauri app ,the react side communicates with python script using websocket and this communication has to be continuous.so i tried to bundle it but then it executes the whole file and i want to make run in parallel

**Reply by smith0022** [reply]
const command = Command.sidecar("bin/python/test-x86_64-unknown-linux-gnu");
const output = command.execute();
outside this function in app.js i imported the command first:
import { Command } from "@tauri-apps/api/shell";

**Comment by smith0022** [comment]
const command = Command.sidecar("bin/python/test-x86_64-unknown-linux-gnu");
const output = command.execute();
outside this function in app.js i imported the command first:
import { Command } from ***@***.***/api/shell";
<https://aka.ms/AAb9ysg>
________________________________
